src/app/pages/add-contact/addContact.component.ts

```diff
@@ -1,9 +1,68 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { BackendService } from '../../services/backend.service';
 
 @Component({
   templateUrl: './addContact.component.html',
   styleUrls: ['./addContact.component.scss']
 })
-export class AddContactComponent {
-  constructor() { }
+export class AddContactComponent implements OnInit {
+
+  addContactDataForm: {
+    name: string,
+    address: string,
+    mobile: string,
+    work: string,
+    home: string,
+    email: string,
+    twitter: string,
+    instagram: string,
+    github: string
+  } = {
+      name: '',
+      address: '',
+      mobile: '',
+      work: '',
+      home: '',
+      email: '',
+      twitter: '',
+      instagram: '',
+      github: ''
+    }
+
+  isEmailInValid: boolean = true;
+  invalidAddContact: boolean = false;
+
+  constructor(
+    private backend: BackendService,
+    private router: Router
+  ) { }
+
+  validateEmail() {
+    const { email } = this.addContactDataForm;
+
+    if (!email) { this.isEmailInValid = true; }
+    else if (email.includes('@')) { this.isEmailInValid = true; }
+    else { this.isEmailInValid = false; }
+  }
+
+  ngOnInit() {
+    return this.backend.contacts()
```

Comment:
Do you need to retrieve all the contacts for a page where you're adding a contact?